app/v1/controllers/ListBenefitsController.scala

```diff
@@ -69,7 +69,7 @@ class ListBenefitsController @Inject()(val authService: EnrolmentsAuthService,
           hateoasResponse <- EitherT.fromEither[Future](
             hateoasFactory
               .wrapList(
-                serviceResponse.responseData,
+                process(serviceResponse.responseData, benefitId),
```

Comment:
Should we even be doing this at the controller layer?  As the logic relates to the models, is it not better that this sits inside the ListBenefitsResponse class with the actual link logic?

Reply:
Nope, we are applying HATEOAS to the response in the controller and it depends on the GET/LIST call. We can move the code to model, but ...

Reply:
but?

Reply:
modifying the model inside a function of the model object is really awful :)